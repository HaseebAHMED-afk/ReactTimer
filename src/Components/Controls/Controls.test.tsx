import { shallow } from 'enzyme';
import React from 'react';
import Controls from './Control';
import setTimeInSeconds from '../Main/Main'

describe('Controls Test Cases', () => {
    let container:any;

    beforeEach(()=> (container = shallow(<Controls time={setTimeInSeconds} />)))

    it('renders controls component', () => {
        expect(container).toMatchSnapshot()
    })

    it('renders all the buttons', () => {
        expect(container.find('button').length).toEqual(3)
    })
})
