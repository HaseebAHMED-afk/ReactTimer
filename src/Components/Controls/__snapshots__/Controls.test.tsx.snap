// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls Test Cases renders controls component 1`] = `
<div
  className="controls-container"
>
  <button
    data-testid="btn-play"
    disabled={false}
    onClick={[Function]}
  >
    Play
  </button>
  <button
    data-testid="btn-stop"
    onClick={[Function]}
  >
    Stop
  </button>
  <button
    data-testid="btn-reset"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
